src/user/proposals/UserProposalDashboard/DashboardContent/DashboardContent.js

```diff
@@ -178,90 +206,121 @@ class DashboardContent extends Component {
-                          {proposalItem.proposalStatus}
-                        </span>
-                      </Button>
-                    ) : (
-                      <Button
-                        variant="primary"
-                        className="status-btn-submitted"
-                        size="sm"
-                      >
-                        <span className="btn-content">
-                          {proposalItem.proposalStatus}
-                        </span>
-                      </Button>
-                    )}
-                  </div>
-                  <div className="proposal-content">
-                    <div className="proposal-details">
-                      {proposalItem.proposalDescription}
+            {displayItems
+              .slice(currentPage * pageSize, (currentPage + 1) * pageSize)
+              .map((proposalItem, index) => {
+                return (
+                  <div className="single-proposal" key={index}>
+                    <div className="user-info">
+                      <div className="image">
+                        <Image src={userIcon2} alt="icon" rounded />
+                      </div>
+                      <div className="img-desc">
+                        <h2>{proposalItem.title}</h2>
+                        <p className="proposal-date">
+                          {proposalItem.createdAt}
+                        </p>
+                      </div>
+                      {proposalItem.proposalStatus === "DRAFT" ? (
+                        <Button
+                          variant="primary"
+                          className="status-btn"
+                          size="sm"
+                        >
+                          <span className="btn-content">
+                            {proposalItem.proposalStatus}
+                          </span>
+                        </Button>
+                      ) : (
+                        <Button
+                          variant="primary"
+                          className="status-btn-submitted"
+                          size="sm"
+                        >
+                          <span className="btn-content">
+                            {proposalItem.proposalStatus}
+                          </span>
+                        </Button>
+                      )}
                     </div>
-                    <div className="proposal-options">
-                      {proposalItem.creator === this.state.userId ? (
+                    <div className="proposal-content">
+                      <div className="proposal-details">
+                        {proposalItem.proposalDescription}
+                      </div>
+                      <div className="proposal-options">
+                        {proposalItem.creator === this.state.userId ? (
+                          <Link
+                            to={{
+                              pathname: "/proposaleditor",
```

Comment:
Also why are sending state from here, is there any particular reason?
I think you can just maintain this proposaleditor state itself using props.
Not sure why you have used, let me know

Reply:
Is there a way to pass props between two components that are not directly related as child/parent components?

Reply:
Not really but here you can use ```to={`/proposaleditor/${proposal._id}`}``` in that component in ```componentDidMount``` use ```check if this.props.params showing you that id or not then use that``` or else you can use ```withRouter``` from react-router-dom and just connect the component with that and do ```this.props.history.push('/..');```

But I guess the present one will not create any issue, so let it be.